src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5825,6 +5846,8 @@ void makeAndAdjustReceiverSlot(BoundExpression receiver)
 
             resultType ??= node.Type?.SetUnknownNullabilityForReferenceTypes();
 
+            UnsplitIfNeeded(resultType);
```

Comment:
>UnsplitIfNeeded(resultType); [](http://example.com/codeflow?start=12&length=28)

Are there tests that demonstrate the effect of this change?

Reply:
Likewise, `ConditionalOperator_17` and `ConditionalOperator_WithUserDefinedConversion_BoolOperand` hit the new assertion without this.